pkg/attestlib/tutorial.md

```diff
@@ -0,0 +1,124 @@
+
+   ### PKIX
+    For PKIX, the `CreateAttestation` method creates a signature and stores it in the Attestation’s `Signature` field. The signed payload is stored in the Attestation’s `SerializedPayload` field.
+    
+   ### JWT
+   The `CreateAttestation` method generates a JWT consisting of three parts: a header, payload, and signature. The following fields are populated in the header:
+   ```
+   {
+       "typ": "JWT",
+       "alg": signature_algorithm,
+       "kid": public_key_id,
+   }
+   ```
+
+   The payload containing the image digest is stored in the payload section. The signature is stored in the signature section. The JWT is serialized and stored in the Attestation’s `Signature` field.
+   
+   
+# Verifying
+
+The Verifier holds a set of PublicKeys. When given an Attestation, the Verifier checks if any of its PublicKeys can verify the Attestation. To verify an Attestation, you must create a Verifier, which is initialized with a set of PublicKeys.
+
+1. Create a slice of PublicKeys.
+
+   Create a PublicKey by calling the appropriate `NewPublicKey` constructor based on the type of Attestation that key should verify. For example, if you have a key that verifies PGP Attestations, call `NewPgpPublicKey`.
+   
+   ### PGP
+   A PGP PublicKey can be created by passing in the raw key material:
+   ```
+   publicKey := `-----BEGIN PGP PUBLIC KEY BLOCK-----
+   `
+   key, err := NewPgpPublicKey([]byte(publicKey))
+   ```
+   
+   ### Raw PKIX / JWT
+   A PKIX or JWT PublicKey can be created by passing in the raw key material, a key ID, and the key’s SignatureAlgorithm (see [list of supported SignatureAlgorithms](https://github.com/grafeas/kritis/blob/master/pkg/attestlib/signature_algorithm.go#L24)). Note that if the key ID is left as an empty string, a key ID will be generated based on the DER of the key material.
+   #### PKIX
+   ```
+   publicKey := `-----BEGIN PUBLIC KEY BLOCK-----
+   `
+   key, err := NewPkixPublicKey([]byte(publicKey), “some-key-id”, EcdsaP256Sha256)
+   ```
+   
+   #### JWT
+   ```
+   publicKey := `-----BEGIN PUBLIC KEY BLOCK-----
+   `
+   key, err := NewJwtPublicKey([]byte(publicKey), “some-key-id”, EcdsaP256Sha256)
+   ```
+
+   
+
+2. Create a Verifier.
+
+   Create a Verifier using the `NewVerifier` constructor, passing in the image digest you are checking and a slice of PublicKeys.
+   
+   ```
+   publicKeys := []PublicKey{ ... }
+   image := "gcr.io/foo@sha256:bar"
+   verifier := NewVerifier(image, publicKeys)
+   ```
+
+3. Call VerifyAttestation.
+
+   Verify an Attestation by passing it to the Verifier’s `VerifyAttestation` method. The method will check that the Verifier holds a public key that verifies the Attestation. It also checks that the signed payload corresponds to the image you are checking. If either step fails, the method returns an error:
+   
```

Comment:
It would be useful to describe how the payload is checked against the image.

Reply:
Done.